File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> res;

        for (int i = 0; i < strs.size(); i++) {
            string s = strs[i];
            vector<int> char_counts(26, 0);

            for (int j = 0; j < s.size(); j++) {
                //cout<<(s[j] - 'a')<<"\n";
                char_counts[s[j] - 'a']++;
            }

            //cout<<char_counts[3]<<"\n";

            string key = "";
            for (int k = 0; k < 26; k++) {
                key += (char_counts[k]);// + "#"; 
            }

            cout<<key<<"\n";
            res[key].push_back(s);
        }

        vector<vector<string>> result;
        for (auto it = res.begin(); it != res.end(); it++) {
            result.push_back(it->second);
        }

        return result;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        res, gr = [], defaultdict(list)
        print(self.get_anagram("bobby"))

        for s in strs:
            gr[self.get_anagram(s)].append(s)
        
        res.extend(gr.values())
        return res
    

    def get_anagram(self, s) -> int:
        co = [0] * 26
        for c in s:
            #            a
            co[ord(c) - 97] += 1

        res = []
        for i in range(26):
            if co[i] != 0:
                res.extend([chr(i + 97), str(co[i])])
        
        return "".join(res)